src/diffusers/loaders.py

```diff
@@ -1083,6 +1128,16 @@ def load_lora_into_text_encoder(cls, state_dict, network_alpha, text_encoder, lo
                             f"{name}.out_proj.lora_linear_layer.down.weight"
                         ] = text_encoder_lora_state_dict.pop(f"{name}.to_out_lora.down.weight")
 
+                    for name, _ in text_encoder_mlp_modules(text_encoder):
+                        for direction in ["up", "down"]:
+                            for layer in ["fc1", "fc2"]:
+                                original_key = f"{name}.{layer}.lora.{direction}.weight"
+                                replacement_key = f"{name}.{layer}.lora_linear_layer.{direction}.weight"
+                                if original_key in text_encoder_lora_state_dict:
+                                    text_encoder_lora_state_dict[replacement_key] = text_encoder_lora_state_dict.pop(
+                                        original_key
+                                    )
+
```

Comment:
If these layers are newly supported in this PR why would they be added to the legacy naming convention conversion code?

Reply:
I went with the legacy convention because the rest of the conversion logic used the legacy convention (which made it match the locality of `_convert_kohya_ss_state_dict`) but I also understand this is a new addition to the legacy part of the codebase so I'll try to directly write it in the new convention with a brief comment about why it doesn't match the actual function (it should be refactored separately I think, maybe even after the proposed #4247).

Reply:
Ah, does _convert_kohya_ss_state_dict currently return keys in the legacy naming convention?

Reply:
I think so.

Reply:
> Ah, does _convert_kohya_ss_state_dict currently return keys in the legacy naming convention?

It does, actually, IMO. This is why integration tests also pass. Otherwise, there would have been assertion problems I think.

Reply:
ok, I think we should follow up then and update the rest of _convert_kohya_ss_state_dict to use the new naming convention